.github/workflows/build.yml

```
@@ -0,0 +1,187 @@
+      with:
+          python-version: '3.8'
+    - name: Install packages
+      run: sudo apt-get -y install zip pandoc man
+    - name: Set version suffix
+      id: version_suffix
+      env:
+        PUSH_VERSION_COMMIT: ${{ secrets.PUSH_VERSION_COMMIT }}
+      if: "env.PUSH_VERSION_COMMIT == ''"
+      run: echo ::set-output name=version_suffix::$(date -u +"%H%M%S")
+    - name: Bump version
+      id: bump_version
+      run: |
+        python devscripts/update-version.py ${{ steps.version_suffix.outputs.version_suffix }}
+        make issuetemplates
+    - name: Push to release
+      id: push_release
+      run: echo ::set-output name=head_sha::$(git rev-parse HEAD)
+    - name: Get Changelog
+      id: get_changelog
+      run: |
+        changelog=$(cat ChangeLog | grep -oPz '(?s)(?<=version ${{ steps.bump_version.outputs.ytdl_version }}\n{2}).+?(?=\n{2,3}version)') || true
+        echo "changelog<<EOF" >> $GITHUB_ENV
+        echo "$changelog" >> $GITHUB_ENV
+        echo "EOF" >> $GITHUB_ENV
+
+    - name: Run Make
+      run: make all tar
+    - name: Get SHA2-256SUMS for youtube-dl
+      id: sha256_bin
+      run: echo "::set-output name=sha256_bin::$(sha256sum youtube-dl | awk '{print $1}')"
+    - name: Get SHA2-256SUMS for youtube-dl.tar.gz
+      id: sha256_tar
+      run: echo "::set-output name=sha256_tar::$(sha256sum youtube-dl.tar.gz | awk '{print $1}')"
+    - name: Get SHA2-512SUMS for youtube-dl
+      id: sha512_bin
+      run: echo "::set-output name=sha512_bin::$(sha512sum youtube-dl | awk '{print $1}')"
+    - name: Get SHA2-512SUMS for youtube-dl.tar.gz
+      id: sha512_tar
+      run: echo "::set-output name=sha512_tar::$(sha512sum youtube-dl.tar.gz | awk '{print $1}')"
+
+    - name: Create Release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ steps.bump_version.outputs.ytdl_version }}
+        release_name: youtube-dl ${{ steps.bump_version.outputs.ytdl_version }}
+        commitish: ${{ steps.push_release.outputs.head_sha }}
+        body: ${{ env.changelog }}
+        draft: false
+        prerelease: false
+    - name: Upload youtube-dl Unix binary
+      id: upload-release-asset
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ steps.create_release.outputs.upload_url }}
+        asset_path: ./youtube-dl
+        asset_name: youtube-dl
+        asset_content_type: application/octet-stream
+    - name: Upload Source tar
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ steps.create_release.outputs.upload_url }}
+        asset_path: ./youtube-dl.tar.gz
+        asset_name: youtube-dl-${{ steps.bump_version.outputs.ytdl_version }}.tar.gz
+        asset_content_type: application/gzip
```

This action is archived.

```suggestion
    - name: Create Release
      id: create_release
      uses: ncipollo/release-action@v1
      with:
        tag: ${{ steps.bump_version.outputs.ytdl_version }}
        name: youtube-dl ${{ steps.bump_version.outputs.ytdl_version }}
        commit: ${{ steps.push_release.outputs.head_sha }}
        body: ${{ env.changelog }}
        draft: false
        prerelease: false
        artifacts: |
          ./youtube-dl,./youtube-dl.tar.gz
        
```

For other inputs see the readme: https://github.com/ncipollo/release-action